scripts/dependency-graph.js

```diff
@@ -1,189 +1,153 @@
 }
 
-const walk = function (tree, onlyOurs) {
-  const annotations = [] // mermaid dependency annotations
-  const dependedBy = {}
+function walk(tree, onlyOurs) {
+  const annotations = [];
+  const dependedBy = {};
 
-  iterate(tree, dependedBy, annotations, onlyOurs)
+  iterate(tree, dependedBy, annotations, onlyOurs);
 
-  const allDeps = new Set(Object.keys(dependedBy))
-  const foundDeps = new Set()
-  const hierarchy = []
+  const allDeps = new Set(Object.keys(dependedBy));
+  const foundDeps = new Set();
+  const hierarchy = [];
 
   if (onlyOurs) {
     while (allDeps.size) {
-      log.silly('SIZE', allDeps.size)
-      const level = []
+      const level = [];
 
       for (const dep of allDeps) {
-        log.silly(dep, '::', [...dependedBy[dep]].join(', '))
-        log.silly('-'.repeat(80))
-
         if (!dependedBy[dep].size) {
-          level.push(dep)
-          foundDeps.add(dep)
+          level.push(dep);
+          foundDeps.add(dep);
         }
       }
 
-      log.silly('LEVEL', level.length)
-      log.silly('FOUND', foundDeps.size)
-
       for (const dep of allDeps) {
         for (const found of foundDeps) {
-          allDeps.delete(found)
-          dependedBy[dep].delete(found)
+          allDeps.delete(found);
+          dependedBy[dep].delete(found);
         }
       }
 
-      log.silly('SIZE', allDeps.size)
-
       if (!level.length) {
-        const remaining = `Remaining deps: ${[...allDeps.keys()]}`
-        throw new Error(`Would do an infinite loop here, need to debug. ${remaining}`)
+        const remaining = `Remaining deps: ${[...allDeps.keys()]}`;
+        throw new Error(`Would do an infinite loop here, need to debug. ${remaining}`);
       }
 
-      hierarchy.push(level.join(', '))
-      log.silly('HIERARCHY', hierarchy.length)
-      log.silly('='.repeat(80))
+      hierarchy.push(level.join(", "));
     }
   }
 
-  return [annotations, hierarchy]
+  return [annotations, hierarchy];
 }
 
-const iterate = function (node, dependedBy, annotations, onlyOurs) {
+function iterate(node, dependedBy, annotations, onlyOurs) {
   if (!dependedBy[node.packageName]) {
-    dependedBy[node.packageName] = new Set()
+    dependedBy[node.packageName] = new Set();
   }
   for (const [name, edge] of node.edgesOut) {
-    if (
-      (!onlyOurs || isOurs(name)) && !node.dev
-    ) {
+    if ((!onlyOurs || isOurs(name)) && !node.dev) {
       if (!dependedBy[node.packageName].has(edge.name)) {
-        dependedBy[node.packageName].add(edge.name)
-        annotations.push(`  ${stripName(node.packageName)}-->${escapeName(edge.name)};`)
+        dependedBy[node.packageName].add(edge.name);
+        annotations.push(`  ${stripName(node.packageName)}-->${escapeName(edge.name)};`);
         if (edge.to) {
-          iterate(edge.to.target, dependedBy, annotations, onlyOurs)
+          iterate(edge.to.target, dependedBy, annotations, onlyOurs);
         }
       }
     }
   }
 }
 
-run(main)
+main();
```

Comment:
run is our built-in script runner wrapper that brings logging and other args parsing to the table. It is important.